bb8/src/inner.rs

```diff
@@ -10,13 +11,36 @@ use tokio::spawn;
 use tokio::time::{interval_at, sleep, timeout, Interval};
 
 use crate::api::{Builder, ConnectionState, ManageConnection, PooledConnection, RunError};
-use crate::internals::{Approval, ApprovalIter, Conn, SharedPool, State};
+use crate::internals::{Approval, ApprovalIter, Conn, SharedPool};
+
+struct SharedPoolInnerStats {
```

Comment:
Why can't this be a part of the `SharedPool`?

Style: code in bb8 follows a top-down style, so definitions should be moved below any code that is using it (in descending order of size/complexity).

Reply:
> Why can't this be a part of the SharedPool?

I guess that was the place were was easily to measure the [`get`](https://github.com/pfreixes/bb8/blob/add_get_stats/bb8/src/inner.rs#L116) contention. Also if we move forward with this PR I would like to also add new attribute for counting the total time waited, something similar to the [`WaitDuration`](https://pkg.go.dev/database/sql#DBStats) exposed by Golang for the `sql.DB` pool. If we go in that way for making this calculation the `SharedPool` will give us the right place for measuring this time, 

> Style: code in bb8 follows a top-down style, so definitions should be moved below any code that is using it (in descending order of size/complexity).
Oks Ill move this to down

Reply:
Also we might want to consider that other attributes like `MaxIdleClosed`, `MaxIdleTimeClosed`, `MaxLifetimeClosed`.

Reply:
> I guess that was the place were was easily to measure the [get](https://github.com/pfreixes/bb8/blob/add_get_stats/bb8/src/inner.rs#L116) contention.

What do you mean? What does the place to measure have to do with the place for storing?

Reply:
> What do you mean? What does the place to measure have to do with the place for storing?

🤦 now i see that I totally digressed to a different subject, sorry. I just stored it into a different class without any specific reason, Im happy to move them as simple attributes of the `SharedPool` and thats all.

Reply:
Now Ive remembered another reason for having all new statistics attributes under its own type, if I do not do this this would force me to warp all of them under the `Arc` object.

If this is not a big issue for you I can do this, otherwise with current proposal we can just share a reference to one single instance of `SharedPoolInnerStatistics`.

Up to you